### ISSUE TITLE
Can not export model to mlflow format locally

### ISSUE THREAD

**Issue by Jeffwan** [ludwig-ai/ludwig#2353]
**Describe the bug**

I try to run `ludwig export_mlflow --model_path results/experiment_run/model/ --output_pat /data/results/mlflow` after the training and I hope to convert ludwig model to MLflow format for future usage. Howevery 

```
ludwig export_mlflow --model_path results/experiment_run/model/ --output_pat /data/results/mlflow
NumExpr defaulting to 8 threads.
███████████████████████
█ █ █ █  ▜█ █ █ █ █   █
█ █ █ █ █ █ █ █ █ █ ███
█ █   █ █ █ █ █ █ █ ▌ █
█ █████ █ █ █ █ █ █ █ █
█     █  ▟█     █ █   █
███████████████████████
ludwig v0.6.dev - Export MLFlow

Model path: results/experiment_run/model/
Output path: /data/results/mlflow


Traceback (most recent call last):
  File "/usr/local/bin/ludwig", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.8/site-packages/ludwig/cli.py", line 166, in main
    CLI()
  File "/usr/local/lib/python3.8/site-packages/ludwig/cli.py", line 66, in __init__
    getattr(self, args.command)()
  File "/usr/local/lib/python3.8/site-packages/ludwig/cli.py", line 136, in export_mlflow
    export.cli_export_mlflow(sys.argv[2:])
  File "/usr/local/lib/python3.8/site-packages/ludwig/export.py", line 337, in cli_export_mlflow
    export_mlflow(**vars(args))
  File "/usr/local/lib/python3.8/site-packages/ludwig/export.py", line 133, in export_mlflow
    export_model(model_path, output_path, registered_model_name)
  File "/usr/local/lib/python3.8/site-packages/ludwig/contribs/mlflow/model.py", line 270, in export_model
    log_model(
  File "/usr/local/lib/python3.8/site-packages/ludwig/contribs/mlflow/model.py", line 207, in log_model
    Model.log(
  File "/usr/local/lib/python3.8/site-packages/mlflow/models/model.py", line 295, in log
    mlflow.tracking.fluent.log_artifacts(local_path, artifact_path)
  File "/usr/local/lib/python3.8/site-packages/mlflow/tracking/fluent.py", line 785, in log_artifacts
    MlflowClient().log_artifacts(run_id, local_dir, artifact_path)
  File "/usr/local/lib/python3.8/site-packages/mlflow/tracking/client.py", line 1001, in log_artifacts
    self._tracking_client.log_artifacts(run_id, local_dir, artifact_path)
  File "/usr/local/lib/python3.8/site-packages/mlflow/tracking/_tracking_service/client.py", line 374, in log_artifacts
    self._get_artifact_repo(run_id).log_artifacts(local_dir, artifact_path)
  File "/usr/local/lib/python3.8/site-packages/mlflow/store/artifact/local_artifact_repo.py", line 48, in log_artifacts
    verify_artifact_path(artifact_path)
  File "/usr/local/lib/python3.8/site-packages/mlflow/store/artifact/artifact_repo.py", line 304, in verify_artifact_path
    raise MlflowException(
mlflow.exceptions.MlflowException: Invalid artifact path: '/data/results/mlflow'. Names may be treated as files in certain cases, and must not resolve to other names when treated as such. This name would resolve to '/data/results/mlflow'
```

**To Reproduce**

Dockerfile

```
FROM ludwigai/ludwig:nightly

RUN apt update -y && apt install -y vim wget tree
RUN pip install mlflow

ENTRYPOINT ["/bin/bash", "-l", "-c"]
```

inside the container

```
wget https://ludwig-ai.github.io/ludwig-docs/0.5/data/rotten_tomatoes.csv
ludwig init_config --dataset /data/rotten_tomatoes.csv --target=recommended --hyperopt=true --output /data/rotten_tomatoes.yaml
ludwig train --config rotten_tomatoes.yaml --dataset /data/rotten_tomatoes.csv

# now I get the model and I want to export to MLFlow format
ludwig export_mlflow --model_path results/experiment_run/model/ --output_path /data/results/mlflow
```

**Expected behavior**

It should not throw error.

**Screenshots**
![image](https://user-images.githubusercontent.com/4739316/183224863-37a3ef7f-8fcc-4fff-b433-a1833e22609d.png)

**Additional context**

I did check the code and notice this code snippet throws the error
https://github.com/mlflow/mlflow/blob/03a6a2e09e1176976338b40c7adb594543b79226/mlflow/store/artifact/artifact_repo.py#L302-L306

however, If we remove prefix `/`. I get other errors and seems it ask me to give protocol but I do want to save it locally,
![image](https://user-images.githubusercontent.com/4739316/183224935-1c84c9e1-af0a-4a80-833a-1dfc5e49650a.png)

**Comment by tgaddair** [comment]
Hey @Jeffwan, I tried reproducing this locally, but haven't been successful yet. I added a test in #2369 to ensure this works as expected. Can you describe your mlflow setup in more detail (version, if you're setting any environment variables, etc.).

**Comment by tgaddair** [comment]
Ah, I missed the repro steps in Docker. I'll try that out next and get back to you.

**Comment by tgaddair** [comment]
Interesting, was able to repro it on the CLI, but not with this simple repro script in python:

```
from ludwig.export import export_mlflow

export_mlflow("results/experiment_run/model", "data/results/mlfow")
```

So looks like this issue is specific to the CLI. I'll try and see what's different here.

**Comment by tgaddair** [comment]
Okay, looks like the issue was that `--registered_model_name` was being set to `mlflow` by default. This was resulting in mlflow looking for an experiment artifact with this name, when none existed.

I added a fix in #2369 to change the default to `None`, so the CLI should work as expected. @Jeffwan please give it a try when you get a chance and let me know if it addresses your issue.

**Comment by Jeffwan** [comment]
@tgaddair I can confirm the nightly version works well with the patch. Thanks a lot!
